### PR TITLE
add :type annotations to all defcustom variables

### PR DESCRIPTION
**Pull request by wbolster**
…so that the customize interface transforms the current spaghetti into pure awesomeness.

### CODE REVIEW

**Comment by wbolster** [comment]
this is much better than the current huge chunk of elisp that shows up without the the `:type` annotations, don't you think? :)

image

File: dumb-jump.el [hunk 1]
```diff
@@ -49,82 +49,100 @@
...
-  :group 'dumb-jump)
+  :group 'dumb-jump
+  :type 'string)
 
 (defcustom dumb-jump-fallback-search
   t
   "If nothing is found with normal search fallback to searching the fallback regex."
-  :group 'dumb-jump)
+  :group 'dumb-jump
+  :type 'string)
```
Comment (jacktasia) [inline dumb-jump.el]: This should be boolean.

**Comment by jacktasia** [comment]
> this is much better than the current huge chunk of elisp that shows up without the the :type annotations, don't you think? :)

I do! Thanks so much for taking the time to add these. I just saw one incorrect type, which I commented on.

File: dumb-jump.el (hunk 1 above)
Comment (wbolster) [inline dumb-jump.el]: fixed

**Comment by jacktasia** [comment]
Thanks!

**Pull request merged**